### PR TITLE
subsys: shell: manual `*` parsing

### LINKED ISSUE #10195: Shell dereferences invalid pointer when printing demo command help
On a shell over uart, typing "demo\r" without any option causes a bad pointer de-refence:

==30417==    at 0x804AD0C: _vprintk (printk.c:196)
==30417==    by 0x804FFA3: shell_fprintf_fmt (shell_fprintf.c:38)
==30417==    by 0x804F7BD: shell_fprintf (shell.c:1409)
==30417==    by 0x804FADC: help_item_print (shell.c:1537)
==30417==    by 0x804FB62: help_options_print (shell.c:1561)
==30417==    by 0x804FEB8: shell_help_print (shell.c:1662)
==30417==    by 0x804EECB: shell_execute (shell.c:1080)
==30417==    by 0x804E87C: shell_state_collect (shell.c:770)
==30417==    by 0x804F69F: shell_process (shell.c:1366)
==30417==    by 0x804F3B5: shell_thread (shell.c:1232)
==30417==  Address 0xa is not stack'd, malloc'd or (recently) free'd

### PR DESCRIPTION
**Pull request by jakub-uC**
This PR fixes: #10195.
Function _vprintk when used cannot parse '*' what as a result
causes dereferencing bad pointer.

This change is replacement for: #10299

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

### CODE REVIEW

File: subsys/shell/shell.c [hunk 1]
```diff
@@ -1533,12 +1533,22 @@ static void help_item_print(const struct shell *shell, const char *item_name,
...
 	}
 
+#if !defined(CONFIG_NEWLIB_LIBC) && !defined(CONFIG_ARCH_POSIX)
 	/* print option name */
 	shell_fprintf(shell, SHELL_NORMAL, "%s%-*s%s:",
 		      tabulator,
 		      item_name_width, item_name,
 		      tabulator);
+#else
+	u16_t tmp = item_name_width - strlen(item_name);;
```
Comment (aescolar) [inline subsys/shell/shell.c]: silly thing: Double ``;``

**Review by aescolar — APPROVED** [review]
I guess that will solve it. I'm guessing you are trying to avoid using a buffer to pre-build the string, so I guess there isn't any obvious nicer way.

**Comment by aescolar** [comment]
@jarz-nordic : I guess you tested it?

**Comment by jarz-nordic** [comment]
@aescolar : Yes I tested it.
I will correct your finding shortly.

File: subsys/shell/shell.c [hunk 2]
```diff
@@ -1533,12 +1533,22 @@ static void help_item_print(const struct shell *shell, const char *item_name,
 		return;
 	}
 
+#if !defined(CONFIG_NEWLIB_LIBC) && !defined(CONFIG_ARCH_POSIX)
```
[the remark this answers is not in the document]
Comment (jarz-nordic) [reply subsys/shell/shell.c]: done

**Review by carlescufi — APPROVED** [review]

**Pull request merged**